docs/enhancements/0002_day0-integration.md

```diff
@@ -0,0 +1,285 @@
+in `osImageURL` then MCO will override the node OS based on the `MachineConfig`
+and reboot - we are going to leverage this mechanisem for the integration with
+KMM.
+
+In a day-0-cluster scenario, the user has applied this `MachineConfig` to tell
+MCO he has modified the OS on the nodes and that it shoudn't override it with
+the RHCOS image from the payload.
+
+### Setting up the integration
+
+A user that wishes to hand over the management of his day-0 kmod to KMM will
+first need to make sure the `MachineConfig` can be reconize by KMM, therefore,
+he would add a `machineconfiguration.openshift.io/kmm-managed: "true"` label to
+it.
+
+KMM will watch for `MachineConfig`s in the cluster containing the
+`machineconfiguration.openshift.io/kmm-managed: "true"` label and create a new
+`ModuleDay0` CR in the cluster for each one of them.
+We will have a different controller for this new CRD in KMM.
+**There should be no more than 1 `ModuleDay0` targeting a `MachineConfigPool`
+(masters/workers).
+If multiple kmods are required they should all be backed in the same image**
+
+Since MCO can only apply `MachineConfig`s changes to an entire `MachineConfigPool`,
+we must require that a `ModuleDay0` will target a `MachineConfigPool` as well.
+MCO ties between the `machineconfiguration.openshift.io/role: worker` label on
+the `MachineConfig` and the `node-role.kubernetes.io/worker: ""` label on the
+node.
+
+A customer can always create his own `MachineConfigPool` and add only 1 worker
+to it but I don't think this should be one of our primary concerns ATM for the
+following reasons
+  * assisted-installer will generate the same ISO for all nodes in the cluster
+    anyway so there is no benefit of having finer grained node selector for
+    `ModuleDay0` when assisted doesn't support such resolution on its own.
+  * if a user wishes to have only some worker nodes modified, he will need to
+    create is own `MachineConfigPool` post cluster installation and then update
+    the `machineconfiguration.openshift.io/role: worker` label of the
+    `MachineConfig` to target the new `MachineConfigPool`.
+    This will make MCO role back the nodes that shouldn't be updated to the
+    payload OS.
+    Using layering for only some of the workers will result in a cluster upgrade
+    that only upgrade the nodes that aren't layered which may cause some
+    inconsistency and confusion
+
+After the `ModuleDay0` was create by KMM, all KMM's features will be used by the
+official KMM's API which is the `ModuleDay0` objects in the cluster.
+
+In case a user will wish to stop the sync between MCO and KMM, he will need to
+remove that label from the `MachineConfig` and KMM will delete its `ModuleDay0`.
+Once done, if the user delete the `MachineConfig` in the cluster then MCO will
+roll-back to the RHCOS image from the payload (without the kmods).
+
+### The `ModuleDay0` CRD
+
+```
+apiVersion: kmm.sigs.x-k8s.io/v1beta1
+kind: ModuleDay0
+spec:
+  osMapping:
```

Comment:
Do we need this `osMapping` if the `containerImage` is to be set in all nodes of the MCP?

Reply:
Nope. This is a good point. I will modify it in an additional commit so it is easier to review (will squash before merge).

Reply:
Fixed as part of https://github.com/rh-ecosystem-edge/kernel-module-management/pull/957/commits/f666db49bc9b8ecac344c3b2d4a967ae60ad8e85